### PR TITLE
Add new function `hitTestSceneObjects` and refactor

### PR DESCRIPTION
**Pull request by macrozone**
this replaces https://github.com/HippoAR/react-native-arkit/pull/35 and is remerged with current master.

Notice that the way properties are passed to the native code has slightly changed:

the NSDictionary passed to the native code now has the exact same shape as the props of the corresponding components. This simplifies the code and makes it easier to maintain

### CODE REVIEW

File: components/ARCapsule.js [hunk 1]
```diff
@@ -57,6 +23,7 @@ ARCapsule.propTypes = {
   shape: PropTypes.shape({
     capR: PropTypes.number,
     height: PropTypes.number,
+    color: PropTypes.string,
```
Comment (qftgtr) [inline components/ARCapsule.js]: color is placed in the shader now

File: ios/RCTARKit.m [hunk 2]
```diff
@@ -12,6 +12,7 @@
 
 @interface RCTARKit () <ARSCNViewDelegate, ARSessionDelegate, UIGestureRecognizerDelegate> {
     RCTPromiseResolveBlock _resolve;
+    BOOL _metal;
```
Comment (qftgtr) [inline ios/RCTARKit.m]: I removed it in the last PR and this is not used anymore.

File: ios/RCTARKit.m [hunk 3]
```diff
@@ -65,6 +66,8 @@ - (instancetype)initWithARView:(ARSCNView *)arView {
         // start ARKit
         [self addSubview:arView];
         [self resume];
+        
+        _metal = YES;
```
Comment (qftgtr) [inline ios/RCTARKit.m]: I removed it in the last PR and this is not used anymore.

File: package.json [hunk 4]
```diff
@@ -1,21 +1,15 @@
 {
-  "name": "react-native-arkit",
+  "name": "@panter/react-native-arkit",
```
Comment (qftgtr) [inline package.json]: please remove `@panter`

Comment (macrozone) [reply package.json]: sure, accident

**Pull request merged**